cep-recipe-jinja.md

```diff
@@ -0,0 +1,125 @@
+# Jinja functions in recipes
+
+The new recipe format has some Jinja functionalities. We want to specify what functions exist and their expected behavior.
+
+## The compiler function
+
+The compiler function is used to stick together a compiler from {lang}_compiler and {lang}_compiler_version
+
+The function looks as follows:
+
+```yaml
+${{ compiler('c') }}
+```
+
+This would pull in the c_compiler and c_compiler_version from the variant config. The compiler function also adds the target_platform to render to something such as:
+
+```
+gcc_linux-64 8.9
+clang_osx-arm64 12
+msvc_win-64 19.29
+```
+
+The function thus evaluates to `{compiler}_{target_platform} {compiler_version}`.
+
+The variant config could look something like:
```

Comment:
I'm not sure "could" is the right verb here. Are we suggesting? Standardizing? If it's not set in stone, why is it useful in this doc? As Cheng proposes, we should standardize the configuration file here.

Reply:
I changed the text, and refrain from specifying _how_ the values are configured in the `variant`.